src/device-state.coffee

```diff
@@ -694,7 +694,8 @@ module.exports = class DeviceState extends EventEmitter
 		@applyInProgress = true
 		new Promise (resolve, reject) =>
 			setTimeout(resolve, delay)
-			@cancelDelay = reject
+			if @cancelDelay
+				@cancelDelay = reject()
```

Comment:
This is a little odd, where we don't know whether the class var is a function or boolean, perhaps this whole workflow should be revisited.